ra-optimisation/search-tree-rao/src/main/java/com/powsybl/openrao/searchtreerao/marmot/Marmot.java

```diff
@@ -94,8 +97,7 @@ public CompletableFuture<InterTemporalRaoResult> run(InterTemporalRaoInput raoIn
         LinearOptimizationResult postTopologicalOptimizationResult = getPostTopologicalOptimizationResult(postTopologicalActionsResults, preventiveTopologicalActions, objectiveFunction);
 
         // if no inter-temporal constraints are defined, the results can be returned
-        // TODO : Add intertemporal constraint check if none violated then return
-        if (raoInput.getGeneratorConstraints().isEmpty()) {
+        if (noInterTemporalConstraint) {
             OpenRaoLoggerProvider.TECHNICAL_LOGS.info("[MARMOT] No inter-temporal constraint provided; no need to re-optimize range actions");
             return CompletableFuture.completedFuture(new InterTemporalRaoResultImpl(initialObjectiveFunctionResult, postTopologicalOptimizationResult, topologicalOptimizationResults));
```

Comment:
There is still a problem here, the postTopologicalOptimizationResult will not contain the cost of the actions.